File: ocmsdk/src/main/java/com/gigigo/orchextra/core/data/api/mappers/elements/ApiElementSectionViewMapper.java
package com.gigigo.orchextra.core.data.api.mappers.elements;

import com.gigigo.orchextra.core.data.api.dto.elements.ApiElementSectionView;
import com.gigigo.orchextra.core.domain.entities.elements.ElementSectionView;
import com.gigigo.ggglib.mappers.ExternalClassToModelMapper;

public class ApiElementSectionViewMapper
        implements ExternalClassToModelMapper<ApiElementSectionView, ElementSectionView> {

    @Override
    public ElementSectionView externalClassToModel(ApiElementSectionView data) {
        ElementSectionView model = new ElementSectionView();

        model.setText(data.getText());
        model.setImageUrl(data.getImageUrl());
        model.setImageThumb(data.getImageThumb());

        return model;
    }
}

File: ocmsdk/src/main/java/com/gigigo/orchextra/core/data/api/mappers/elements/ApiElementSectionViewMapper.kt
package com.gigigo.orchextra.core.data.api.mappers.elements

import com.gigigo.ggglib.mappers.ExternalClassToModelMapper
import com.gigigo.orchextra.core.data.api.dto.elements.ApiElementSectionView
import com.gigigo.orchextra.core.domain.entities.elements.ElementSectionView

class ApiElementSectionViewMapper : ExternalClassToModelMapper<ApiElementSectionView, ElementSectionView> {

  override fun externalClassToModel(data: ApiElementSectionView): ElementSectionView {
    val model = ElementSectionView()
    with(model) {
      text = data.text
      imageUrl = data.imageUrl
      imageThumb = data.imageThumb
    }
    return model
  }
}
